OpenRA.Game/Network/Connection.cs

```diff
@@ -252,8 +257,31 @@ void NetworkConnectionReceive()
 			}
 		}
 
+		void Ack(byte[] buf)
+		{
+			var stream = new MemoryStream(buf);
+			var frameReceived = stream.ReadInt32();
+			stream.ReadByte();
+			var framesToAck = stream.ReadInt16();
+			var timestep = stream.ReadInt16();
+
+			var ms = new MemoryStream(4 + awaitingAckPackets.Take(framesToAck).Sum(i => i.Length));
```

Comment:
Is it already possible to use `Span<>` in the dotnet version that is used?

In #19019 i propose to pass a Span<> over the frame data to avoid splitting up a packet into frames using a separate memory stream. To avoid data received from a socket being copied multiple few times before it's processed.

Reply:
Better to leave it in a followup if we want to do it.